### PR TITLE
sinkmanager(ticdc): implement sink manager

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5928

### What is changed and how it works?

Implement sink manager and add tests.

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/processor/sinkmanager/manager_impl.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+				continue
+			}
+			// If redo log is disabled, we use the barrier ts as the upper bound of the fetch tableSinkTask.
+			// Because can not exceed the barrier ts.
+			barrierTs := m.lastBarrierTs.Load()
+			upperBoundPos := sorter.Position{
+				StartTs:  barrierTs - 1,
+				CommitTs: barrierTs,
+			}
+			canBeAdvance := slowestTableProgress.nextLowerBoundPos.Compare(upperBoundPos) == -1 ||
```
Comment (hicqu) [inline cdc/processor/sinkmanager/manager_impl.go:116]: Seems it's better:
```go
checkAdvance := slowestTableProgress.nextLowerBoundPos.Compare(upperBoundPos)
if checkAdvance == -1 || checkAdvance == 0 {
}
```

Comment (hi-rustin) [reply cdc/processor/sinkmanager/manager_impl.go]: Good idea! Thanks!

File: pkg/config/server_config.go [hunk 2]
```diff
@@ -46,6 +46,9 @@ const (
 	// It is larger than TiDB's txn-entry-size-limit.
 	// We can't set it to a larger value without risking oom in incremental scenarios.
 	DefaultTableMemoryQuota = 10 * 1024 * 1024 // 10 MB
+
+	// DefaultChangefeedMemoryQuota is the default memory quota for each changefeed.
+	DefaultChangefeedMemoryQuota = 10 * 1024 * 1024 * 1024 // 10 GB
```
Comment (hicqu) [inline pkg/config/server_config.go:51]: Seems it's too large. How about 1G?

File: cdc/processor/sinkmanager/manager.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/cdc/processor/pipeline"
+)
+
+// Manager manages the sink of a processor.
+// NOTICE: All methods of Manager are thread-safe.
+type Manager interface {
+	// UpdateBarrierTs updates the barrier ts of all tables in the sink manager.
+	// It is used to advance the progress if RedoLog is not enabled.
```
Comment (hicqu) [inline cdc/processor/sinkmanager/manager.go:25]: Seems barrierTs is used when RedoLog is **enabled**. If RedoLog isn't enabled, only `tableResolvedTs` will be used.

Comment (hicqu) [reply cdc/processor/sinkmanager/manager.go:25]: I suggest to remove the line. BarrierTs is also used if redo is enabled.

File: cdc/processor/sinkmanager/manager_impl.go [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+					zap.String("namespace", m.changefeedID.Namespace),
+					zap.String("changefeed", m.changefeedID.ID),
+					zap.Int64("tableID", tableID))
+				// Maybe the table sink is removed by the processor.(Scheduled the table to other nodes.)
+				// So we do **not** need add it back to the heap.
+				continue
+			}
+			// We use the barrier ts as the upper bound of the fetch tableSinkTask.
+			// Because it can not exceed the barrier ts.
+			barrierTs := m.lastBarrierTs.Load()
```
[the remark this answers is not in the document]
Comment (hi-rustin) [reply cdc/processor/sinkmanager/manager_impl.go]: Updated!

**Review by hicqu — APPROVED** [review]

File: cdc/processor/sinkmanager/manager_impl.go [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+// RemoveTable removes a table(TableSink) from the sink manager.
+func (m *ManagerImpl) RemoveTable(tableID model.TableID) error {
+	tableSink, ok := m.tableSinks.Load(tableID)
+	if !ok {
+		log.Panic("Table sink not found when removing table",
+			zap.String("namespace", m.changefeedID.Namespace),
+			zap.String("changefeed", m.changefeedID.ID),
+			zap.Int64("tableID", tableID))
+	}
+	err := tableSink.(*tableSinkWrapper).close(m.ctx)
```
Comment (nongfushanquan) [inline cdc/processor/sinkmanager/manager_impl.go:293]: if a table is removed,  what will happen for the sink worker in the backend

Comment (hi-rustin) [reply cdc/processor/sinkmanager/manager_impl.go:294]: Good question! For now, nothing happened — no more progress advance.

**Pull request merged**